### ISSUE TITLE
DRF 3.10 deprecated decorators

### ISSUE THREAD

**Issue by n2ygk** [django-json-api/django-rest-framework-json-api#657]
The drfmaster builds are currently failing due to a deprecation described in https://github.com/encode/django-rest-framework/pull/6687:
```
@detail_route/@list_route (replaced by @action)
```

Here's a snippet showing the error:
```
____________________ ERROR collecting example/urls_test.py _____________________
ImportError while importing test module '/home/travis/build/django-json-api/django-rest-framework-json-api/example/urls_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
example/urls_test.py:4: in <module>
    from .api.resources.identity import GenericIdentity, Identity
example/api/resources/identity.py:4: in <module>
    from rest_framework.decorators import detail_route, list_route
E   ImportError: cannot import name 'detail_route' from 'rest_framework.decorators' (/home/travis/build/django-json-api/django-rest-framework-json-api/.tox/py37-df20-django22-drfmaster/lib/python3.7/site-packages/rest_framework/decorators.py)
```

**Comment by n2ygk** [comment]
Looks like a pretty easy fix. Set `detail=True` for detail, `False` for list:
```
def action(methods=None, detail=None, url_path=None, url_name=None, **kwargs):
    """
    Mark a ViewSet method as a routable action.

    Set the `detail` boolean to determine if this action should apply to
    instance/detail requests or collection/list requests.
    """

```

**Comment by n2ygk** [comment]
Unfortunately the `@action` decorator is relatively new, so simply replacing it breaks with drf<=3.7

**Comment by n2ygk** [comment]
The workaround wasn't terrible though. PR coming.

**Comment by sliverc** [comment]
This got fixed with #664
